contracts/src/TeleporterTokenSource.sol

```diff
@@ -543,12 +551,15 @@ abstract contract TeleporterTokenSource is
         } else if (bridgeMessage.messageType == BridgeMessageType.REGISTER_DESTINATION) {
             RegisterDestinationMessage memory payload =
                 abi.decode(bridgeMessage.payload, (RegisterDestinationMessage));
+            require(
+                payload.sourceTokenDecimals == tokenDecimals,
+                "TeleporterTokenSource: invalid token decmials"
```

Comment:
```suggestion
                "TeleporterTokenSource: invalid token decimals"
```

Reply:
fyi see related discussion here: https://github.com/ava-labs/teleporter-token-bridge/pull/142#discussion_r1608699742

Having said all that, I'm happy to do whatever you think is Right.

Reply:
Yeah, I think it makes more sense to check this in _registerDestination, and also pass the whole struct

Reply:
addressed in https://github.com/ava-labs/teleporter-token-bridge/pull/142/commits/38b0066a32ab9fa5d96e121e78313cfa2238c371